### PR TITLE
Simple E-filing through a DA interview Guide added

### PR DESCRIPTION
**Pull request by BryceStevenWilley**
Added a simple guide that walks through the basic parts of e-filing in a docassemble interview.
Some additional documents that I want to write that would be useful (and might feel like missing components of this guide):
* advanced e-filing (i.e. what you'd find in a production interview)
* concepts, like filing documents, codes, case parties, etc.

Also added some redirects, so e-filing can be in it's own section.

I'll merge after I edit it tomorrow morning, but any comments welcome, I haven't yet had time to edit it myself.

### CODE REVIEW

File: docs/efiling/efiling_through_docassemble.md [hunk 1]
```diff
@@ -0,0 +1,190 @@
+---
+id: efiling_through_docassemble
+title: |
+  E-filing through Docassemble
+sidebar_label: Filing + Docassemble
```
Comment (nonprofittechy) [inline docs/efiling/efiling_through_docassemble.md:5]:
```suggestion
sidebar_label: Integrating Docassemble interviews with e-filing
```

File: docs/efiling/efiling_through_docassemble.md [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+to be embedded into your interview.
+
+## Before starting
+
+Before starting this guide, you should have:
+
+* a working docassemble interview, preferably using the AssemblyLine.
+* If you don't have one yet, follow [the steps to author your interview](https://suffolklitlab.org/docassemble-AssemblyLine-documentation/docs/planning).
+* contacted us for access to the EFSP server. We aren't yet making access publicly available, but will include an interest form here when we do.
+
```
Comment (nonprofittechy) [inline docs/efiling/efiling_through_docassemble.md:28]:
```suggestion
* Installed the docassemble,EFSP package on your server via the Package Management menu

```

File: docs/efiling/efiling_through_docassemble.md [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+* the `case_subtype` (if necessary)
+* individual's `party_type`
+* several for the filing itself:
+  * `document_type`
+  * `filing_component`
+  * `tyler_filing_type`
+  * `filing_parties`
+  * `filing_action`
+
+To find the right codes, you can navigate through the URLs provided by visiting <https://efile.suffolklitlab.org:9000/jurisdictions/illinois/codes>.
```
Comment (nonprofittechy) [inline docs/efiling/efiling_through_docassemble.md:88]: These links don't display anything for me in Firefox--looks like just an empty page.

**Review by nonprofittechy — APPROVED** [review]
Looks good! I think adding a small glossary of some of the obscure Tyler codes would be useful on this page, but it's not a blocker.

Another thought is to consider using some tables to make an easier quick reference of the other variables that need to be defined to e-file. It could include the name, the type, and the purpose to facilitate a quick lookup.

**Comment by nonprofittechy** [comment]
Other high level notes:
- I like the separate section. Clearly e-filing is going to be a central draw for our project and it's going to be relevant to people who don't use the rest of our project
- Watch consistency of terms--"set", "has", "include" - I think you mostly want to say "ask questions" and "define". Avoid using synonyms so it's clear what the author needs to do in each case
- You use the term "attribute" pretty broadly but I think since sometimes they are top level variables, just using "variable" throughout would be clearer.

I think the mix of small examples and top level sections works pretty well as is. But I think you could embrace the "this is an annotated interview example" even more--sometimes you introduce big concepts and sometimes you're zeroing in on a specific example. It's not always clear in each section which one is going to appear. You could think of breaking each section into a subsection. Overview and small annotated example, in a consistent pattern, so readers can better follow the documentation.

That said this is very short and readable as is, so additional signposting is not critical.

File: docs/efiling/efiling_through_docassemble.md (hunk 2 above)
Comment (BryceStevenWilley) [reply docs/efiling/efiling_through_docassemble.md:28]: I like that step, but I'll wouldn't consider it to be a pre-req, we can walk them through it in this guide.

File: docs/efiling/efiling_through_docassemble.md (hunk 3 above)
Comment (BryceStevenWilley) [reply docs/efiling/efiling_through_docassemble.md:88]: You mean that page itself? That's expected, I haven't deployed the latest proxy to prod yet, but I'll do that today and double check all of the links.

File: docs/efiling/efiling_through_docassemble.md [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+
+* `tyler_case_category = '7306'`: Adoption
+* `tyler_case_type = '25361'`: Adoption
+* no `case_subtype`
+* `party_type = '7067' or '20640'`: Adoptive parent and Adoptive child
+* for the filing itself:
+  * `tyler_filing_type='29730'`: Application
+  * `document_type = '5766'`: Non-confidential
+  * `filing_component = '332'`: Lead document
+
```
[the remark this answers is not in the document]
Comment (BryceStevenWilley) [reply docs/efiling/efiling_through_docassemble.md:111]: By terms, you mean like `filing_component` and `document_type` right (as opposed to the numbers, or the specific names of the codes?) I was planning on a concept explainer page, thought there might be so much that a full glossary page is required.

File: docs/efiling/efiling_through_docassemble.md [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+### Give all of the info that the EFSP needs to file your doc
+
+Now that we have all of the codes that we want, we need to set up the logic of the interview.
+
+You should have a `users` and a `other_parties` list of individual. At the very least, each of these individuals should have a first name. Note that if only a first name is provided, then the person is assumed to be a business.
+
+For each object in `users` and in `other_parties`, you need to set the following attributes:
```
[the remark this answers is not in the document]
Comment (BryceStevenWilley) [reply docs/efiling/efiling_through_docassemble.md:118]: I get what your trying to say, but in this case, you probably shouldn't ask the user to define `party_type` and `is_new`, most of the time you'll be able to define it yourself in the logic, without asking the user specifically about this. I will work on the wording overall though.

File: docs/efiling/efiling_through_docassemble.md (hunk 3 above)
Comment (BryceStevenWilley) [reply docs/efiling/efiling_through_docassemble.md]: Tyler seems to have buggered up their TLS certs during a migration, and I can't deploy a new version when their site is broken, so this'll have to just be wrong for a bit.

**Comment by BryceStevenWilley** [comment]
I think that the biggest thing to fix is the "big concept" vs "specific example" that you mentioned. I'm going to try to write a concept guide and glossary soon, and when that's in, I can take out the bigger discussions from this guide, which should streamline and make this guide clearer. I think it's good for now though, so I'm going to merge.

**Pull request merged**